qiita_db/util.py

```diff
@@ -1219,3 +1219,159 @@ def supported_filepath_types(artifact_type):
+                        study_id = qiita.study.study_id) AS artifact_biom_ts,
+                (SELECT array_agg(DISTINCT visibility) FROM qiita.artifact
+                    LEFT JOIN qiita.visibility USING (visibility_id)
+                    WHERE study_id=qiita.study.study_id)
+                    AS artifacts_visibility,
+                (SELECT array_agg(publication_doi ORDER BY publication_doi)
+                    FROM qiita.study_publication
+                    WHERE study_id=qiita.study.study_id) AS publication_doi,
+                (SELECT array_agg(name ORDER BY email) FROM qiita.study_users
+                    LEFT JOIN qiita.qiita_user USING (email)
+                    WHERE study_id=qiita.study.study_id) AS shared_with_name,
+                (SELECT array_agg(email ORDER BY email) FROM qiita.study_users
+                    LEFT JOIN qiita.qiita_user USING (email)
+                    WHERE study_id=qiita.study.study_id) AS shared_with_email
+                FROM qiita.study
+                LEFT JOIN qiita.study_person ON (
+                    study_person_id=principal_investigator_id)
+                WHERE study_id IN %s"""
+        qdb.sql_connection.TRN.add(sql, [tuple(study_ids)])
+        infolist = []
+        refs = {}
+        for info in qdb.sql_connection.TRN.execute_fetchindex():
+            info = dict(info)
+
+            # publication info
+            if info['publication_doi'] is not None:
+                info['pmid'] = get_pubmed_ids_from_dois(
+                    info['publication_doi']).values()
+            else:
+                info['publication_doi'] = []
+                info['pmid'] = []
+
+            # visibility
+            info["status"] = infer_status([info['artifacts_visibility']])
+            del info['artifacts_visibility']
+
+            # pi info
+            info["pi"] = (info['pi_email'], info['pi_name'])
+            del info["pi_email"]
+            del info["pi_name"]
+
+            # shared with
+            info['shared'] = []
+            if info['shared_with_name']:
+                for name, email in zip(info['shared_with_name'],
+                                       info['shared_with_email']):
+                    if not name:
+                        name = email
+                    info['shared'].append((email, name))
+            del info["shared_with_name"]
+            del info["shared_with_email"]
+
+            info['proc_data_info'] = []
+            if build_samples and info['artifact_biom_ids']:
+                to_loop = zip(
+                    info['artifact_biom_ids'], info['artifact_biom_dts'],
+                    info['artifact_biom_ts'], info['artifact_biom_params'])
+                for artifact_id, dt, ts, params in to_loop:
+                    proc_info = {'processed_date': str(ts)}
+                    proc_info['pid'] = artifact_id
+                    proc_info['data_type'] = dt
+                    del params['input_data']
```

Comment:
This is hard coded and it is not ensured that the parameter name will be `input_data`. I think it should check for the parameters that are artifacts and remove them.

Also, all the lines below here, are hard-coded for target gene - and I don't think they work in case that the reference used is UNITE (AFAIK there is no tree associated with it).

What is the outcome once we start adding the other processing pipelines? This may not work for a table resulting from HUMANN2.

What is going to happen with the BIOM artifacts uploaded by the user? They may not have this information.